fabric-model-loading-api-v1/src/client/java/net/fabricmc/fabric/api/client/model/loading/v1/WrapperUnbakedItemModel.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.client.model.loading.v1;
+
+import com.mojang.serialization.MapCodec;
+
+import net.minecraft.client.render.item.model.ItemModel;
+
+/**
+ * A simple implementation of {@link ItemModel.Unbaked} that delegates all method calls to the {@link #wrapped} field.
+ * Implementations must set the {@link #wrapped} field somehow.
```

Comment:
Question: I assume there is a good reason why wrapped isnt final, and the mod is expected to set the field.

Generally we tend not expose fields as part of the API, would an `abstract ItemModel.Unbaked getWrapped()` be more suitable for these classes.

Reply:
Fabric API had the `ForwardingBakedModel` class for the longest time, which used the design of having a protected, non-final field. Any new model wrapper classes use the same design. The field could be final, or the class could be removed entirely, but it's so simple that the field may as well be non-final to cover a couple extra use cases so the user doesn't need to copy the class.